unity/Assets/Scripts/CollisionListener.cs

```diff
@@ -0,0 +1,120 @@
+
+using UnityStandardAssets.Characters.FirstPerson;
+
+public class CollisionListener : MonoBehaviour
+{
+    //references to the joints of the mid level arm
+    [SerializeField]
+    private bool CascadeCollisionEventsToParent = true;
+    //track what was hit while arm was moving
+    public class StaticCollision
+    {
+        public GameObject gameObject;
+
+        public SimObjPhysics simObjPhysics;
+
+         //indicates if gameObject a simObject
+        public bool isSimObj {
+            get { return simObjPhysics != null; }
+        }
+    }
+    private StaticCollision staticCollided;
+
+    private HashSet<Collider> activeColliders = new HashSet<Collider>();
+
+    public void RegisterCollision(Collider col, bool notifyParent = true) {
+        activeColliders.Add(col);
+        if (notifyParent && this.transform.parent != null) {
+            foreach (var listener in this.transform.parent.GetComponentsInParent<CollisionListener>()) {
+                listener.RegisterCollision(col, notifyParent);
+            }
+        }     
+    }
+
+    public void DeregisterCollision(Collider col, bool notifyParent = true) {
+        activeColliders.Remove(col);
+        if (notifyParent && this.transform.parent != null) {
+            foreach (var listener in this.transform.parent.GetComponentsInParent<CollisionListener>()) {
+                listener.DeregisterCollision(col, notifyParent);
+            }
+        } 
+    }
+
+    public void Reset(bool? notifyParent = null) {
+        activeColliders.Clear();
+        if (notifyParent.GetValueOrDefault(this.CascadeCollisionEventsToParent) && this.transform.parent != null) {
+            foreach (var listener in this.transform.parent.GetComponentsInParent<CollisionListener>()) {
+                listener.Reset(notifyParent);
+            }
+        } 
+    }
+
+    public void OnTriggerExit(Collider col)
+    {
+        DeregisterCollision(col, CascadeCollisionEventsToParent);
+    }
+
+    public void OnTriggerStay(Collider col)
+    {
+        this.
+        RegisterCollision(col, CascadeCollisionEventsToParent);
+    }
+
+    public void OnTriggerEnter(Collider col)
+    {
+        RegisterCollision(col, CascadeCollisionEventsToParent);
+    }
+
+    private StaticCollision ColliderToStaticCollision(Collider col) {
+        StaticCollision sc = null;
+        if(col.GetComponentInParent<SimObjPhysics>())
+            {
+                //how does this handle nested sim objects? maybe it's fine?
+                SimObjPhysics sop = col.GetComponentInParent<SimObjPhysics>();
+                if(sop.PrimaryProperty == SimObjPrimaryProperty.Static)
+                {
+
+                    if(!col.isTrigger)
+                    {
+                        // #if UNITY_EDITOR
+                        // Debug.Log("Collided with static sim obj " + sop.name);
+                        // #endif
+                        sc = new StaticCollision();
+                        sc.simObjPhysics = sop;
```

Comment:
In the previous code only a simObj or gameObject was stored on the StaticCollision.  I don't think it matters too much, but it forces us to only do the null check where we check for a simobj first and then a gameobject in a certain order.  We only do that during debugging, so I don't think it changes the actual logic.

Reply:
I wanted to simplify the check to not have an if that looked like `if (sc.simObjPhysics != null && sc.gameObject == null) { // then it is a simobject` which we had, so basically now if it is registered as a "StaticCollision" the gameObject will always be non-Null and there is a `isSimObject` field for quick checking if we collided against a SimObject.

The way to check if it is/is not a simObject you only call `sc.isSimObject`, and you avoid the null checking for the game object so it is the same one check, I just think is cleaner to have less nulls around and simplifies the above case. 
Does this make sense?